server/utils/courseParticipationUtils.ts

```diff
@@ -0,0 +1,89 @@
+import DateUtils from './dateUtils'
+import StringUtils from './stringUtils'
+import type { CourseParticipationWithName } from '@accredited-programmes/models'
+import type { GovukFrontendSummaryListRowWithValue } from '@accredited-programmes/ui'
+import type { GovukFrontendSummaryList } from '@govuk-frontend'
+
+export default class CourseParticipationUtils {
+  static summaryListOptions(courseParticipationWithName: CourseParticipationWithName): GovukFrontendSummaryList {
+    return {
+      card: {
+        title: {
+          text: courseParticipationWithName.name,
+        },
+      },
+      rows: CourseParticipationUtils.summaryListRows(courseParticipationWithName),
+    }
+  }
+
+  private static summaryListRows(
+    courseParticipationWithName: CourseParticipationWithName,
+  ): Array<GovukFrontendSummaryListRowWithValue> {
+    const summaryListRows: Array<GovukFrontendSummaryListRowWithValue> = []
+
+    summaryListRows.push({
+      key: { text: 'Programme name' },
+      value: { text: courseParticipationWithName.name },
+    })
+
+    if (courseParticipationWithName.setting) {
```

Comment:
How would you feel about extracting this out to private methods e.g. `courseSettingRow(courseParticipationWithName)` and `courseOutcomeRow`? Might make this a bit easier to scan.

Reply:
I did wonder about this before. I've added a possible way of doing this in a fixup commit: https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/221/commits/7260430319fb0c86538d3fb269966ec85b1dda77

This is using methods with side effects, which feels a little off in a utility class, but maybe it's okay. Initially I was attempting to have the methods just return the row for it to be added in the `summaryListRows` method, but that requires a lot of extra typing/type assertion. The conditional logic narrows the typing, which allows us to construct the row object without needing more explicit and complicated type assertion, so it's easier to keep the conditional logic in the same context as the pushes to the summary list rows array to avoid type gymnastics

Reply:
Have reverted to original approach as discussed. We might simplify this later if we're moving towards always showing the rows even when there's no data